omm2caom2/tests/test_omm_name.py

```diff
@@ -66,79 +66,124 @@
-    assert OmmName(file_name='C121212_00001_TEST.fits.gz').is_valid()
-    assert OmmName(file_name='C121212_00001_FOCUS.fits.gz').is_valid()
-    assert OmmName(
-        file_name='C121121_J024345.57-021326.4_K_SCIRED.fits.gz'
-    ).is_valid()
+    original_collection = StorageName.collection
+    original_pattern = StorageName.collection_pattern
+    original_scheme = StorageName.scheme
+    try:
+        StorageName.collection = 'OMM'
+        StorageName.collection_pattern = OmmName.OMM_NAME_PATTERN
+        StorageName.scheme = SCHEME
+
+        assert OmmName(file_name='C121212_00001_SCI.fits.gz').is_valid()
+        assert not OmmName(file_name='c121212_00001_SCI.fits.gz').is_valid()
+        assert OmmName(file_name='C121212_00001_CAL.fits.gz').is_valid()
+        assert not OmmName(file_name='c121212_00001_CAL.fits.gz').is_valid()
+        assert OmmName(
+            file_name='C121212_domeflat_K_CALRED.fits.gz'
+        ).is_valid()
+        assert not OmmName(
+            file_name='C121212_DOMEFLAT_K_CALRED.fits.gz'
+        ).is_valid()
+        assert OmmName(
+            file_name='C121212_sh2-132_J_old_SCIRED.fits.gz'
+        ).is_valid()
+        assert OmmName(
+            file_name='C121212_J0454+8024_J_SCIRED.fits.gz'
+        ).is_valid()
+        assert OmmName(file_name='C121212_00001_TEST.fits.gz').is_valid()
+        assert OmmName(file_name='C121212_00001_FOCUS.fits.gz').is_valid()
+        assert OmmName(
+            file_name='C121121_J024345.57-021326.4_K_SCIRED.fits.gz'
+        ).is_valid()
+
+        test_subject = OmmName(file_name='C121212_00001_SCI.fits')
+        assert test_subject.is_valid()
+        assert test_subject.obs_id == 'C121212_00001'
+        test_subject = OmmName(file_name='C121212_00001_SCI.fits.gz')
+        assert test_subject.is_valid()
+        assert test_subject.obs_id == 'C121212_00001'
+        test_subject = OmmName(
+            file_name='C121212_00001_SCI.fits.gz',
+        )
+        assert test_subject.is_valid()
+        assert test_subject.obs_id == 'C121212_00001'
+        assert (
+            test_subject.file_uri == f'{SCHEME}:OMM/C121212_00001_SCI.fits'
+        ), 'wrong file uri'
 
-    test_subject = OmmName(file_name='C121212_00001_SCI.fits')
-    assert test_subject.is_valid()
-    assert test_subject.obs_id == 'C121212_00001'
-    test_subject = OmmName(file_name='C121212_00001_SCI.fits.gz')
-    assert test_subject.is_valid()
-    assert test_subject.obs_id == 'C121212_00001'
-    test_subject = OmmName(
-        fname_on_disk='C121212_00001_SCI.fits',
-        file_name='C121212_00001_SCI.fits.gz',
-    )
-    assert test_subject.is_valid()
-    assert test_subject.obs_id == 'C121212_00001'
-    assert (
-        test_subject.file_uri == 'ad:OMM/C121212_00001_SCI.fits.gz'
-    ), 'wrong file uri'
+        with pytest.raises(CadcException):
+            test_subject = OmmName._add_extensions('C121212_00001_SCI')
```

Comment:
I'm not sure whether this was the intention, but this succeeds when the first of the calls raises the exception.

Reply:
Thanks for catching this - I've split the cases out into separate tests now.